components/automate-ui/package.json

```diff
@@ -15,7 +15,7 @@
     "copy-ui-lib:fonts": "cp -r ../chef-ui-library/dist/collection/assets/fonts/ src/assets/fonts",
     "build": "ng build",
     "build:prod": "npm run build -- --prod",
-    "test": "ng test --watch=false --code-coverage --source-map=false --configuration=no_auth",
+    "test": "/bin/bash -c 'exec 5>&1; t1=$(ng test --watch=false --code-coverage --source-map=false --configuration=no_auth | tee /dev/fd/5; exit ${PIPESTATUS[0]}); if [ $? -ne 0 ]; then exit 1; else echo $t1 | grep 'WARN:' >/dev/null; test $? -ne 0; fi'",
```

Comment:
Lots of things going on here and, alas, you cannot put comments in a JSON file:
1. `/bin/bash -c` -- The build machine is using `sh`; need to use `bash` because of `PIPESTATUS`
2. `exec 5>&1 ... tee /dev/fd/5` -- Want to capture the `ng test` output to reuse it (rather than have to re-run the command) so sending to an unused stream and redirecting that to `stdout` (stream 1) where it can be captured into the `t1` variable.
3. `${PIPESTATUS[0]}` -- Want to check the exit code of `ng test`, which is in a pipe, so need this to grab the exit code of the first pipe component rather than the final one. 
4. `if [ $? -ne 0 ]; then exit 1` -- If that reported non-zero, exit with non-zero.
5. `echo $t1 | grep... ` -- Check for any egregious warnings and...
6. `test $? -ne 0` -- Return a non-zero exit code if found (because if found `$?` will be 0 and the `test` inverts that result).

(Have a deja vu feeling? I used this technique previously on the `ng lint` line though it has was removed when no longer needed--see https://github.com/chef/a2/pull/5338)